programs/server/Server.cpp

```diff
@@ -1740,6 +1740,15 @@ try
     LOG_INFO(log, "Loading metadata from {}", path_str);
 
     LoadTaskPtrs load_metadata_tasks;
+
+    // Make sure that if exception is thrown during startup async, new async loading jobs are not going to be called.
+    // This is important for the case when exception is thrown from loading of metadata with `async_load_databases = false`
+    // to avoid simultaneously running table startups and destructing databases.
+    SCOPE_EXIT_SAFE(
+        LOG_INFO(log, "Stopping AsyncLoader.");
+        global_context->getAsyncLoader().stopAndDoNotWait(); // Note that currently running jobs will proceed
```

Comment:
But does it actually avoid potential race conditions? Maybe we should wait for threads in the thread pool to finish current jobs as well

Reply:
Yes, probably it would be better, I was worried about introducing another wait, maybe deadlocking something. But waiting would look even closer to what we had with ThreadPool. I'll fix